test/third-party/follow-redirect-http.test.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @jest-environment jsdom
+ */
+import { https } from 'follow-redirects'
+import { httpsAgent, HttpServer } from '@open-draft/test-server/http'
+import { ClientRequestInterceptor } from '../../src/interceptors/ClientRequest'
+import type { HttpRequestEventMap } from '../../src/glossary'
+import { waitForClientRequest } from '../helpers'
+
+const resolver = jest.fn<never, Parameters<HttpRequestEventMap['request']>>()
+
+const interceptor = new ClientRequestInterceptor()
+interceptor.on('request', resolver)
+
+const server = new HttpServer((app) => {
+  app.post('/resource', (req, res) => {
+    res.status(200).send('hello from the server')
+  })
+})
+
+beforeAll(async () => {
+  await server.listen()
+  interceptor.apply()
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+afterAll(async () => {
+  interceptor.dispose()
+  await server.close()
+})
+
+test('intercepts a POST request issued by "follow-redirects"', async () => {
+  const { address } = server.https
+  const payload = JSON.stringify({ todo: 'Buy the milk' })
+
+  const req = https.request({
```

Comment:
I understand that it may feel unnecessary to assert that `follow-redirects` actually perform a redirect request but we should still do so. In this assertion, we're making sure that Interceptors don't interfere with that behavior, and that the redirects still happen.